KubeArmor/core/kubeUpdate.go

```diff
@@ -1168,6 +1168,460 @@ func (dm *KubeArmorDaemon) WatchSecurityPolicies() {
+
+	if event.Type == "ADDED" {
+		new := true
+		for _, policy := range dm.SecurityPolicies {
+			if policy.Metadata["namespaceName"] == secPolicy.Metadata["namespaceName"] && policy.Metadata["policyName"] == secPolicy.Metadata["policyName"] {
+				new = false
+				break
+			}
+		}
+		if new {
+			dm.SecurityPolicies = append(dm.SecurityPolicies, secPolicy)
+		}
+	} else if event.Type == "MODIFIED" {
+		for idx, policy := range dm.SecurityPolicies {
+			if policy.Metadata["namespaceName"] == secPolicy.Metadata["namespaceName"] && policy.Metadata["policyName"] == secPolicy.Metadata["policyName"] {
+				dm.SecurityPolicies[idx] = secPolicy
+				break
+			}
+		}
+	} else if event.Type == "DELETED" {
+		for idx, policy := range dm.SecurityPolicies {
+			if policy.Metadata["namespaceName"] == secPolicy.Metadata["namespaceName"] && policy.Metadata["policyName"] == secPolicy.Metadata["policyName"] {
+				dm.SecurityPolicies = append(dm.SecurityPolicies[:idx], dm.SecurityPolicies[idx+1:]...)
+				break
+			}
+		}
+	}
+
+	dm.SecurityPoliciesLock.Unlock()
+
+	dm.Logger.Printf("Detected a Container Security Policy (%s/%s/%s)", strings.ToLower(event.Type), secPolicy.Metadata["namespaceName"], secPolicy.Metadata["policyName"])
+
+	appArmorAnnotations := map[string]string{}
+	appArmorAnnotations[containername] = "kubearmor_" + containername
+
+	if event.Type == "ADDED" {
+		dm.RuntimeEnforcer.UpdateAppArmorProfiles(containername, "ADDED", appArmorAnnotations)
+
+		newPoint := tp.EndPoint{}
+
+		newPoint.NamespaceName = secPolicy.Metadata["namespaceName"]
+		newPoint.EndPointName = containername
+		newPoint.PolicyEnabled = tp.KubeArmorPolicyEnabled
+		newPoint.Identities = []string{"namespaceName=" + secPolicy.Metadata["namespaceName"]}
+		newPoint.ProcessVisibilityEnabled = true
+		newPoint.FileVisibilityEnabled = true
+		newPoint.NetworkVisibilityEnabled = true
+		newPoint.CapabilitiesVisibilityEnabled = true
+
+		newPoint.Containers = []string{}
+		newPoint.AppArmorProfiles = []string{"kubearmor_" + containername}
+
+		newPoint.SecurityPolicies = append(newPoint.SecurityPolicies, secPolicy)
+
+		if cfg.GlobalCfg.Policy {
+			// update security policies
+			dm.Logger.UpdateSecurityPolicies("ADDED", newPoint)
+
+			if dm.RuntimeEnforcer != nil && newPoint.PolicyEnabled == tp.KubeArmorPolicyEnabled {
+				// enforce security policies
+				dm.RuntimeEnforcer.UpdateSecurityPolicies(newPoint)
+			}
+		}
+	} else {
+		dm.RuntimeEnforcer.UpdateAppArmorProfiles(containername, "DELETED", appArmorAnnotations)
+
+		dm.EndPointsLock.Lock()
+		for idx, endPoint := range dm.EndPoints {
+			if secPolicy.Metadata["namespaceName"] == endPoint.NamespaceName && secPolicy.Metadata["podName"] == endPoint.EndPointName {
+				// remove endpoint
+				dm.EndPoints = append(dm.EndPoints[:idx], dm.EndPoints[idx+1:]...)
+				break
+			}
+		}
+		dm.EndPointsLock.Unlock()
+	}
+}
+
+// WatchContainerSecurityPolicies Function
+func (dm *KubeArmorDaemon) WatchContainerSecurityPolicies() {
```

Comment:
Do we need this function?

Reply:
No, it was added as a wrapper function for testing purpose. Will remove it :+1: